src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -1681,8 +1681,19 @@ public int get() {
 
         public void endRecovery() throws ElasticsearchException {
             store.decRef();
-            onGoingRecoveries.decrementAndGet();
+            int left = onGoingRecoveries.decrementAndGet();
             assert onGoingRecoveries.get() >= 0 : "ongoingRecoveries must be >= 0 but was: " + onGoingRecoveries.get();
+            if (left == 0) {
+                try {
+                    flush(new Engine.Flush().type(Flush.Type.COMMIT_TRANSLOG));
```

Comment:
I think both can be left at their defaults? `force=false` `waitIfOngoing=false`. It means there is already an ongoing flush or there is nothing to do. In both cases, no need for an extra flush to reduce the translog.